main.js

```diff
@@ -0,0 +1,85 @@
+const { InstanceBase, Regex, runEntrypoint, InstanceStatus, TCPHelper } = require('@companion-module/base')
+const UpgradeScripts = require('./upgrades')
+const UpdateActions = require('./actions')
+const UpdateFeedbacks = require('./feedbacks')
+const UpdateVariableDefinitions = require('./variables')
+const presets = require('./presets')
+
+class ModuleInstance extends InstanceBase {
+	constructor(internal) {
+		super(internal)
+		Object.assign(this, { ...presets })
+	}
+
+	async init(config) {
+		this.config = config
+
+		//this.updateStatus(InstanceStatus.Ok)
+
+		if (this.socket) {
+			this.socket.destroy()
+			delete this.socket
+		}
+
+
+		if (this.config.host) {
+			this.updateStatus(InstanceStatus.Connecting)
+			this.socket = new TCPHelper(this.config.host, 23)
+
+			this.socket.on('status_change', (status, message) => {
+				this.updateStatus(status, message)
+			})
+
+			this.socket.on('error', (err) => {
+				this.updateStatus(InstanceStatus.ConnectionFailure, err.message)
+				this.log('error', 'Network error: ' + err.message)
+			})
+
+			
+		} else {
+			this.updateStatus(InstanceStatus.BadConfig)
+		}
+
+		this.updateActions() // export actions
+		this.updateFeedbacks() // export feedbacks
+		this.updateVariableDefinitions() // export variable definitions
+		this.initPresets()
+	}
+
+	// When module gets deleted
+	async destroy() {
+		this.log('debug', 'destroy')
+	}
```

Comment:
Socket should be closed on destroy as well

Reply:
fixed -- added socket destroy to the destroy function, moved the TCP connect code into its own function, calling that new function from init, and in the configUpdated (if the host was changed)